db/blip_sync_context.go

```diff
@@ -258,11 +258,24 @@ func (bsc *BlipSyncContext) handleChangesResponse(sender *blip.Sender, response
 				}
 			}
 
+			var responseCallback func(resp *blip.Message)
+			if bsc.postSendRevisionResponseCallback != nil {
+				responseCallback = func(resp *blip.Message) {
+					if resp.Type() == blip.ResponseType {
+						bsc.postSendRevisionResponseCallback(seq.String())
+					} else {
+						respBody, _ := response.Body()
+						base.WarnfCtx(bsc.loggingCtx, "error %s in response to rev: %s", response.Properties["Error-Code"], respBody)
+						// TODO: What do we want to do with SGR2 checkpointing in this case? By never calling the callback, we're effectively preventing all future sequences from being checkpointed, but don't have a way to retry or recover.
```

Comment:
See PR description for open question around this.

Reply:
We should still be invoking postSendRevisionResponseCallback - there are valid reasons for an error to be returned by the remote (conflicts and sync function rejection).  I also think it's valid to treat all errors that are returned by the remote SG as non-recoverable for now - any retry (e.g. for high load on the remote) should be happening on the remote before returning an error.  The exception might be if the remote returns a 503 - we could consider a future enhancement to support retry in that case.